Fails on first request, everytime

When building locally, I get `` and the following stack trace every time I start or restart the server – only upon the first request, the second request goes through fine:

```
server Starting ch.epfl.scala.index.server.Server.main()
[success] Total time: 2 s, completed Jul 1, 2016 2:28:31 PM
3. Waiting for source changes... (press enter to interrupt)
server [ERROR] [07/01/2016 14:28:58.196] [scaladex-akka.actor.default-dispatcher-4] [akka.actor.ActorSystemImpl(scaladex)] Error during processing of request HttpRequest(HttpMethod(GET),http://localhost:8080/,List(Host: localhost:8080, Connection: keep-alive, Upgrade-Insecure-Requests: 1, User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36, Accept: text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8, Accept-Encoding: gzip, deflate, sdch, Accept-Language: en-US, en;q=0.8, Cookie: _ga=GA1.1.196777477.1456161474; cookieconsent_dismissed=yes; _sessiondata=D7B60B18F7896E1A40AC712AFE33DA86740FA03A-1467979396142-x%7B%22repos%22%3A%5B%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scala%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22community-builds%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22healy-article%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22progfun-stats%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22pnwscala.org%22%7D%2C%7B%22organization%22%3A%22gzm0%22%2C%22repo%22%3A%22scala-lang%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22cv%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scala-lang%22%7D%2C%7B%22organization%22%3A%22dickwall%22%2C%22repo%22%3A%22scala%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22sbt-coursera%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22heathermiller.github.io%22%7D%2C%7B%22organization%22%3A%22felixmulder%22%2C%22repo%22%3A%22dotty%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scacs%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22janvitek.github.io%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22docs%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scala-dataflow%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22parcolls-extended%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22menthor%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scala-pickling%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22distributed%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22play-with-scalajs-example%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22dep%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22reactive-stats%22%7D%2C%7B%22organization%22%3A%22dickwall%22%2C%22repo%22%3A%22scala.github.com%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22f-p%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22dissertation%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22papers-documents%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22sbt-example-spores%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22papers%22%7D%2C%7B%22organization%22%3A%22heathermiller%22%2C%22repo%22%3A%22scala-records%22%7D%5D%2C%22user%22%3A%7B%22login%22%3A%22heathermiller%22%2C%22name%22%3A%22Heather+Miller%22%2C%22avatarUrl%22%3A%22https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F687163%3Fv%3D3%22%7D%7D; _refreshtoken=5ki7p0pbvq294hns:6uqvlmv6nsg7m5kilns15inssj1e2b9sfc8usnalk97mjd3dvu98bluraqbf77eb; XSRF-TOKEN=v5nj44i9m5jtg0ckechgqaapan7hg5q5faelqba7phvukgu7sa8riurf198llpjp, Timeout-Access: <function1>),HttpEntity.Strict(none/none,ByteString()),HttpProtocol(HTTP/1.1))
server Failed to execute phase [query], all shards failed
server 	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onFirstPhaseResult(AbstractSearchAsyncAction.java:206)
server 	at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:152)
server 	at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:46)
server 	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:855)
server 	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:833)
server 	at org.elasticsearch.transport.TransportService$4.onFailure(TransportService.java:387)
server 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:39)
server 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
server 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
server 	at java.lang.Thread.run(Thread.java:745)
server 
```

--- a/data/src/main/scala/ch.epfl.scala.index.data/elastic/SeedElasticSearch.scala
+++ b/data/src/main/scala/ch.epfl.scala.index.data/elastic/SeedElasticSearch.scala
@@ -10,7 +10,6 @@ import maven.PomsReader
 import com.sksamuel.elastic4s._
 import ElasticDsl._
 import mappings.FieldType._
-import org.elasticsearch.cluster.health.ClusterHealthStatus
 
 import scala.concurrent.duration._
 import scala.concurrent.{Await, ExecutionContext}
@@ -22,39 +21,6 @@ class SeedElasticSearch(implicit val ec: ExecutionContext) extends ProjectProtoc
     val exists = Await.result(esClient.execute { indexExists(indexName)}, Duration.Inf).isExists()
     val target = indexName / collectionName
 
-    def blockUntilCount(expected: Int): Unit = {
-      blockUntil(s"Expected count of $expected") {
-        () =>
-          expected <= {
-            val res = esClient.execute {
-              search
-                .in(indexName / collectionName)
-                .query(matchAllQuery)
-                .size(0)
-            }.await.totalHits
-            println(res)
-            res
-          }
-
-      }
-    }
-    def blockUntilGreen(): Unit = {
-      blockUntil("Expected cluster to have green status") { () =>
-        esClient.execute {
-          get cluster health
-        }.await.getStatus == ClusterHealthStatus.GREEN
-      }
-    }
-    def blockUntil(explain: String)(predicate: () => Boolean): Unit = {
-      var backoff = 0
-      var done = false
-      while (backoff <= 16 && !done) {
-        if (backoff > 0) Thread.sleep(200L * backoff)
-        backoff = backoff + 1
-        done = predicate()
-      }
-      require(done, s"Failed waiting on: $explain")
-    }
 
 
     if (!exists) {

--- a/data/src/main/scala/ch.epfl.scala.index.data/elastic/package.scala
+++ b/data/src/main/scala/ch.epfl.scala.index.data/elastic/package.scala
@@ -1,11 +1,12 @@
 package ch.epfl.scala.index
 package data
 
+import com.sksamuel.elastic4s.ElasticDsl.{get, _}
 import model._
 import release._
-
 import org.elasticsearch.common.settings.Settings
 import com.sksamuel.elastic4s._
+import org.elasticsearch.cluster.health.ClusterHealthStatus
 import source.Indexable
 import org.json4s._
 import org.json4s.native.Serialization
@@ -46,4 +47,52 @@ package object elastic extends ProjectProtocol {
 
   val indexName = "scaladex"
   val collectionName = "artifacts"
+
+  private def blockUntil(explain: String)(predicate: () => Boolean): Unit = {
+
+    var backoff = 0
+    var done = false
+    while (backoff <= 16 && !done) {
+      if (backoff > 0) Thread.sleep(200L * backoff)
+      backoff = backoff + 1
+      done = predicate()
+    }
+    require(done, s"Failed waiting on: $explain")
+  }
+
+  def blockUntilYellow(): Unit = {
+
+    import ElasticDsl._
+
+    blockUntil("Expected cluster to have green status") { () =>
+      esClient.execute {
+        get cluster health
+      }.await.getStatus == ClusterHealthStatus.YELLOW
+    }
+  }
+
+  def blockUntilCount(expected: Int): Unit = {
+    blockUntil(s"Expected count of $expected") {
+      () =>
+        expected <= {
+          val res = esClient.execute {
+            search
+              .in(indexName / collectionName)
+              .query(matchAllQuery)
+              .size(0)
+          }.await.totalHits
+          println(res)
+          res
+        }
+
+    }
+  }
+
+  def blockUntilGreen(): Unit = {
+    blockUntil("Expected cluster to have green status") { () =>
+      esClient.execute {
+        get cluster health
+      }.await.getStatus == ClusterHealthStatus.GREEN
+    }
+  }
 }

--- a/server/src/main/scala/ch.epfl.scala.index.server/Server.scala
+++ b/server/src/main/scala/ch.epfl.scala.index.server/Server.scala
@@ -2,10 +2,10 @@ package ch.epfl.scala.index
 package server
 
 import model._
-import data.cleanup.SemanticVersionParser
 import model.misc.UserInfo
 import model.release.SemanticVersion
-
+import data.cleanup.SemanticVersionParser
+import data.elastic._
 
 import akka.http.scaladsl._
 import akka.http.scaladsl.model._
@@ -182,6 +182,9 @@ object Server {
         }
       }
     }
+
+    /** wait for elastic to start */
+    blockUntilYellow()
 
     Await.result(Http().bindAndHandle(route, "0.0.0.0", 8080), 20.seconds)
 

--- a/server/src/main/scala/ch.epfl.scala.index.server/Server.scala
+++ b/server/src/main/scala/ch.epfl.scala.index.server/Server.scala
@@ -183,7 +183,7 @@ object Server {
       }
     }
 
-    /** wait for elastic to start */
+    /* wait for elastic to start */
     blockUntilYellow()
 
     Await.result(Http().bindAndHandle(route, "0.0.0.0", 8080), 20.seconds)

